### ISSUE TITLE
ACDC for taskchain wf creating more jobs than the original

### ISSUE THREAD

**Issue by julianbadillo** [dmwm/WMCore#6187]
This request:
jbadillo_task_TOP-RunIIWinter15wmLHE-00029__v1_T_150818_092315_7241
Had 37/200 LHE jobs failed, and I created an ACDC
jbadillo_ACDC_task_TOP-RunIIWinter15wmLHE-00029__v1_T_150831_095330_7255
With same splitting as the original workflow, however, it created 2K LHE jobs 730K GEN-SIM jobs (10x the original).
@amaltaro

**Comment by amaltaro** [comment]
sigh... ACDCs give so much trouble lately, which makes me wonder why we support resubmission for MC from scratch (?) Maybe we should transform this specific case in something like an extension workflow instead, which would also solve the problem when we define a smaller job splitting.

**Comment by amaltaro** [comment]
seems to be happening to ReDigi as well when the job splitting is changed (?) 
elog
https://cms-logbook.cern.ch/elog/Workflow+processing/21924

**Comment by julianbadillo** [comment]
Another case:
 https://cms-logbook.cern.ch/elog/Workflow+processing/22505
Instead of recovering 21/200 LHE jobs, it created 2K LHE jobs

**Comment by amaltaro** [comment]
Another one just created 5M jobs.
https://cmsweb.cern.ch/couchdb/_utils/document.html?reqmgr_workload_cache/jen_a_ACDC_task_SMP-Summer12WMLHE-00022__v1_T_160113_203956_3290

It's time to _patch_ or really fix this annoying issue. On it

**Comment by amaltaro** [comment]
Found the reason for this huge number of jobs created, while investigating the previous workflow.
The problem comes mainly from the acdc doc that is uploaded to the server, when a job fails, it gets the number of events in that file from wmbs, instead of uploading *only* the amount of events that that job was meant to process.

In this example, it means:
 - there were 30502 failed jobs to recover, so let's call it "NumberOfFiles": 30502
 - each of those files has "events": 50000
 - this gives us a total "NumberOfEvents": 1525100000
 - we use only EventAwareLumiBased algo for input data
 - and in this example it was 300 evts per job, so "Jobs": 5083667 (NumberOfEvents/300)
basically this line:
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WorkQueue/Policy/Start/ResubmitBlock.py#L70

It wouldn't happen if we use LumiBased, because them the acdcserver has the correct runLumi mask uploaded. Anyways, it has to work with EventAwareLumiBased.

I still don't know how to properly fix it here (well, that would be uploading the correct NumberOfEvents, though I'm not sure we can put our hands in that info...).

**Comment by amaltaro** [comment]
And one case for MonteCarlo with EventBased spltting:
https://cms-logbook.cern.ch/elog/Workflow+processing/23555

to be investigated  soon...

**Comment by amaltaro** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [ACDC for taskchain wf creating more jobs than the original](https://trello.com/c/TQ8nErEl/1-acdc-for-taskchain-wf-creating-more-jobs-than-the-original)

**Comment by amaltaro** [comment]
Also high priority for 2 years, but since it's a bug, I'll keep it for a bit longer.
What surprises me is that this hasn't happened anymore in production, not that I'm aware of, at least...
